connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -98,6 +104,12 @@ public String toString() {
     // int32 and want the smaller width), we use an otherwise invalid field name in the cast spec to track this.
     private static final String WHOLE_VALUE_CAST = null;
 
+    /**
+     * Known logical types that can be cast meaningfully to primitive target type.
+     */
+    private static final Set<String> LOGICAL_TYPES = new HashSet<>(
+        Arrays.asList(Date.LOGICAL_NAME, Time.LOGICAL_NAME, Timestamp.LOGICAL_NAME));
```

Comment:
Is there a reason not to include support for casting `DECIMAL` to other primitives? Yes, this might be lossy, but it also would help in cases where there would be no loss of precision.

Reply:
The implementation is a bit nuanced because the underlying representation is just a byte[] representing the BigInt. Plus, I was concerned about data loss during the cast. If we are fine with the data loss though, I can add it.

Reply:
@rhauch actually, casting DECIMAL *just works* because the logical representation BigDecimal extends Number and can safely be passed down to the castToInt* methods without going through fromLogical. Going to update comments to say as much.